release-notes/7.0/preview/api-diff/preview2/Microsoft.WindowsDesktop.App/7.0-preview2_System.Windows.Forms.md

```diff
@@ -0,0 +1,192 @@
+# System.Windows.Forms
+
+``` diff
+ namespace System.Windows.Forms {
+     public class Button : ButtonBase, IButtonControl {
+-        public new event EventHandler DoubleClick;
++        public new event EventHandler? DoubleClick;
+-        public new event MouseEventHandler MouseDoubleClick;
++        public new event MouseEventHandler? MouseDoubleClick;
+     }
+     public abstract class ButtonBase : Control {
+-        public Image Image { get; set; }
++        public Image? Image { get; set; }
+-        public ImageList ImageList { get; set; }
++        public ImageList? ImageList { get; set; }
+-        public new event EventHandler AutoSizeChanged;
++        public new event EventHandler? AutoSizeChanged;
+-        public new event EventHandler ImeModeChanged;
++        public new event EventHandler? ImeModeChanged;
+     }
+     public class DataObject : IDataObject, IDataObject {
+-        public virtual Stream GetAudioStream();
++        public virtual Stream? GetAudioStream();
+-        public virtual object GetData(string format);
++        public virtual object? GetData(string format);
+-        public virtual object GetData(string format, bool autoConvert);
++        public virtual object? GetData(string format, bool autoConvert);
+-        public virtual object GetData(Type format);
++        public virtual object? GetData(Type format);
+-        public virtual Image GetImage();
++        public virtual Image? GetImage();
+-        public virtual void SetData(object data);
++        public virtual void SetData(object? data);
+-        public virtual void SetData(string format, bool autoConvert, object data);
++        public virtual void SetData(string format, bool autoConvert, object? data);
+-        public virtual void SetData(string format, object data);
++        public virtual void SetData(string format, object? data);
+-        public virtual void SetData(Type format, object data);
++        public virtual void SetData(Type format, object? data);
+-        int System.Runtime.InteropServices.ComTypes.IDataObject.DAdvise(ref FORMATETC pFormatetc, ADVF advf, IAdviseSink pAdvSink, out int pdwConnection);
++        int IDataObject.DAdvise(ref FORMATETC pFormatetc, ADVF advf, IAdviseSink pAdvSink, out int pdwConnection);
+-        void System.Runtime.InteropServices.ComTypes.IDataObject.DUnadvise(int dwConnection);
++        void IDataObject.DUnadvise(int dwConnection);
+-        int System.Runtime.InteropServices.ComTypes.IDataObject.EnumDAdvise(out IEnumSTATDATA enumAdvise);
++        int IDataObject.EnumDAdvise(out IEnumSTATDATA? enumAdvise);
+-        IEnumFORMATETC System.Runtime.InteropServices.ComTypes.IDataObject.EnumFormatEtc(DATADIR dwDirection);
++        IEnumFORMATETC IDataObject.EnumFormatEtc(DATADIR dwDirection);
+-        int System.Runtime.InteropServices.ComTypes.IDataObject.GetCanonicalFormatEtc(ref FORMATETC pformatetcIn, out FORMATETC pformatetcOut);
```

Comment:
@carlossanlop , can you update your tool to exclude these changes ( removed explicit namespace info from runtime)? I do not see any value in mentioning them as API changes. To me, its misleading.

Reply:
In general, should explicit interface implementation methods be shown at all?

Reply:
It's a good question.
Today the docs document explicit interface implementations, e.g.: https://docs.microsoft.com/dotnet/api/system.windows.forms.dataobject.system-runtime-interopservices-comtypes-idataobject-dadvise

Reply:
The tool had a bug. EIIs should only show up if they are newly added. In this case, the tool was printing an incomplete DocId in one side, and a full DocId in the other.

Please take a look again at the file.